File: sdk/python/examples/multimodal_input.py
"""Send text and a local image as structured input parts."""

import argparse
import asyncio
from pathlib import Path

from dotcraft import DotCraft, LocalOptions, local_image_part, text_part


async def main() -> None:
    parser = argparse.ArgumentParser(description=__doc__)
    parser.add_argument("workspace")
    parser.add_argument("image")
    args = parser.parse_args()

    workspace_path = str(Path(args.workspace).resolve())
    image_path = str(Path(args.image).resolve())
    async with await DotCraft.connect_local(LocalOptions(workspace_path=workspace_path)) as dotcraft:
        thread = await dotcraft.threads.start(user_id="sdk-example")
        result = await thread.run([
            text_part("Describe the important information in this image."),
            local_image_part(image_path),
        ])
        print(result.text)


if __name__ == "__main__":
    asyncio.run(main())

File: sdk/typescript/samples/applications/multimodal-input.ts
import { resolve } from "node:path";

import { DotCraft, localImagePart, textPart } from "@dotcraft/sdk";

const [workspaceArgument, imageArgument] = process.argv.slice(2);
if (!workspaceArgument || !imageArgument) {
  throw new Error("Usage: multimodal-input <workspace-path> <image-path>");
}

const workspacePath = resolve(workspaceArgument);
const imagePath = resolve(imageArgument);
const dotcraft = await DotCraft.local({ workspacePath });
try {
  const thread = await dotcraft.threads.start({ userId: "sdk-example" });
  const result = await thread.run([
    textPart("Describe the important information in this image."),
    localImagePart(imagePath),
  ]);
  console.log(result.text);
} finally {
  await dotcraft.close();
}
